Modules/socketmodule.c

```diff
@@ -2107,6 +2107,21 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
             straddr = PyBytes_AS_STRING(args);
             if (setbdaddr(straddr, &_BT_HCI_MEMB(addr, bdaddr)) < 0)
                 return 0;
+#elif defined(__FreeBSD__)
+            _BT_HCI_MEMB(addr, family) = AF_BLUETOOTH;
+            if (!PyBytes_Check(args)) {
+                PyErr_Format(PyExc_OSError, "%s: "
+                             "wrong node format", caller);
+                return 0;
+            }
+            const char *straddr = PyBytes_AS_STRING(args);
+            size_t len = PyBytes_GET_SIZE(args);
+            if (len >= sizeof(_BT_HCI_MEMB(addr, node))) {
```

Comment:
I think that == is still valid, and you should use `strncpy`.

Test also that the bytes object does not contain embedded nulls.

Reply:
> I think that == is still valid, and you should use strncpy.

Done. I wasn't sure since the manpage doesn't say anything about null character.

> Test also that the bytes object does not contain embedded nulls.

Ok, done.